### ISSUE TITLE
Scrollable.of() was changed on master without deprecation warning

### ISSUE THREAD

**Issue by matthew-carroll** [flutter/flutter#114724]
I've been informed that `Scrollable.of()` on `master` has been changed from returning a `Scrollable?` to returning a `Scrollable` and throwing an exception of no such ancestor is found.

To obtain a `Scrollable?` that might not exist, we must now called `Scrollable.maybeOf()`.

This is a breaking change without any deprecation opportunity. At the moment, `super_editor` is broken on `master` because we call `Scrollable.of()` in situations where there may not be an ancestor `Scrollable`. We handle that situation. But now, Flutter throws an exception.

If we switch to `Scrollable.maybeOf()` that would break all of our clients on `stable` and `beta`, because that method doesn't exist on `stable` or `beta`. We're forced to choose between moving to the latest on `master` or continuing to support our clients on `stable` and beta`.

I don't see why this change is worth all the damage in the first place. Why is a try/catch better than a null check? There's always a possibility that a desired ancestor doesn't exist in the tree. Regardless, there should have been multiple releases with a deprecation warning before Flutter possibly broke all calls to `Scrollable.of()` in the ecosystem.

If I'm missing something about this change, please let me know.

Here's the PR that introduced the change: https://github.com/flutter/flutter/pull/114120

I don't see a linked write-up in that PR description, so I don't know the motivation for this change. But it seems both unnecessary, and more tedious than the existing approach.

CC @gspencergoog

**Comment by goderbauer** [comment]
The breaking change is documented here: https://docs.flutter.dev/release/breaking-changes/supplemental-maybeOf-migration

**Comment by matthew-carroll** [comment]
For the record, the majority of my issue ticket was about an absence of a deprecation period. Nobody addressed that in any way. I'm disappointed with how this was handled.

**Comment by stuartmorgan** [comment]
> Regardless, there should have been multiple releases with a deprecation warning before Flutter possibly broke all calls to `Scrollable.of()` in the ecosystem.

Flutter's documented policy does not require this, and is clear about the fact that it's not the common path for API changes:

https://docs.flutter.dev/resources/compatibility

>> We will, on occasion, deprecate certain APIs rather than outright break them overnight.

**Comment by matthew-carroll** [comment]
> Flutter's documented policy does not require this

Hence, the word "should", not the word "must". The team **should** have deprecated this method.

I wrote a response to expand on my original point, but I decided to delete it. Both responses on this thread are essentially passive aggressive. The team clearly lacks empathy for the consequences of such breakages. I'll leave you to it. My disappointment continues.
